shared/src/bundles/users/validation-schemas/user-update-profile.validation-schema.ts

```diff
@@ -0,0 +1,98 @@
+import { z } from 'zod';
+
+import { UserValidationMessage, UserValidationRule } from '../enums/enums.js';
+import { UnicodePattern } from './constants/constants.js';
+
+const userUpdateProfile = z
+    .object({
+        fullName: z.union([
+            z
+                .string()
+                .trim()
+                .regex(UnicodePattern.FULLNAME_PATTERN, {
+                    message: UserValidationMessage.FULLNAME_WRONG,
+                })
+                .min(UserValidationRule.FULLNAME.MIN_LENGTH, {
+                    message: UserValidationMessage.FULLNAME_LENGTH,
+                })
+                .max(UserValidationRule.FULLNAME.MAX_LENGTH, {
+                    message: UserValidationMessage.FULLNAME_LENGTH,
+                })
+                .nullable(),
+            z.literal(''),
+        ]),
+        username: z.union([
+            z
+                .string()
+                .trim()
+                .regex(UnicodePattern.NICKNAME_PATTERN, {
+                    message: UserValidationMessage.NICKNAME_WRONG,
+                })
+                .min(UserValidationRule.NICKNAME.MIN_LENGTH, {
+                    message: UserValidationMessage.NICKNAME_LENGTH,
+                })
+                .max(UserValidationRule.FULLNAME.MAX_LENGTH, {
+                    message: UserValidationMessage.FULLNAME_LENGTH,
+                })
+                .nullable(),
+            z.literal(''),
+        ]),
+        dateOfBirth: z.union([
+            z
+                .string()
+                .regex(
+                    UnicodePattern.BIRTHDATE_PATTERN,
+                    UserValidationMessage.BIRTHDATE_FORMAT,
+                )
+                .nullable(),
+            z.literal(''),
```

Comment:
zod has built-in datetime strings validation

Reply:
It has, but I need only to accept one specific format of date which is dd/mm/yyyy and according to zod documentation the built-in validation of date will pass for all other date formats (for example "2023-01-10")